code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -505,39 +509,52 @@
-	var/datum/gas_mixture/cooling_remove = cooling_port.remove(0.05 * cooling_port.total_moles())
+	var/datum/gas_mixture/cooling_removed = cooling_port.remove(0.05 * cooling_port_total_moles)
+	cooling_removed_heat_capacity = cooling_removed.heat_capacity()
 	//Cooling of the moderator gases with the cooling loop in and out the core
-	if(moderator_internal.total_moles() > 0)
-		var/coolant_temperature_delta = cooling_remove.temperature - moderator_internal.temperature
-		var/cooling_heat_amount = (1 - (1 - HIGH_EFFICIENCY_CONDUCTIVITY) ** delta_time) * coolant_temperature_delta * (cooling_remove.heat_capacity() * moderator_internal.heat_capacity() / (cooling_remove.heat_capacity() + moderator_internal.heat_capacity()))
-		cooling_remove.temperature = max(cooling_remove.temperature - cooling_heat_amount / cooling_remove.heat_capacity(), TCMB)
-		moderator_internal.temperature = max(moderator_internal.temperature + cooling_heat_amount / moderator_internal.heat_capacity(), TCMB)
-
-	else if(internal_fusion.total_moles() > 0)
-		var/coolant_temperature_delta = cooling_remove.temperature - internal_fusion.temperature
-		var/cooling_heat_amount = (1 - (1 - METALLIC_VOID_CONDUCTIVITY) ** delta_time) * coolant_temperature_delta * (cooling_remove.heat_capacity() * internal_fusion.heat_capacity() / (cooling_remove.heat_capacity() + internal_fusion.heat_capacity()))
-		cooling_remove.temperature = max(cooling_remove.temperature - cooling_heat_amount / cooling_remove.heat_capacity(), TCMB)
-		internal_fusion.temperature = max(internal_fusion.temperature + cooling_heat_amount / internal_fusion.heat_capacity(), TCMB)
-	cooling_port.merge(cooling_remove)
+	if(moderator_internal_total_moles > 0)
+		var/coolant_temperature_delta = cooling_removed.temperature - moderator_internal.temperature
+		var/cooling_heat_amount = (1 - (1 - HIGH_EFFICIENCY_CONDUCTIVITY) ** delta_time) * coolant_temperature_delta * (cooling_removed_heat_capacity * moderator_internal_heat_capacity / (cooling_removed_heat_capacity + moderator_internal_heat_capacity))
+		cooling_removed.temperature = max(cooling_removed.temperature - cooling_heat_amount / cooling_removed_heat_capacity, TCMB)
+		moderator_internal.temperature = max(moderator_internal.temperature + cooling_heat_amount / moderator_internal_heat_capacity, TCMB)
+
+	else if(internal_fusion_total_moles > 0)
+		var/coolant_temperature_delta = cooling_removed.temperature - internal_fusion.temperature
+		var/cooling_heat_amount = (1 - (1 - METALLIC_VOID_CONDUCTIVITY) ** delta_time) * coolant_temperature_delta * (cooling_removed_heat_capacity * internal_fusion_heat_capacity / (cooling_removed_heat_capacity + internal_fusion_heat_capacity))
+		cooling_removed.temperature = max(cooling_removed.temperature - cooling_heat_amount / cooling_removed_heat_capacity, TCMB)
+		internal_fusion.temperature = max(internal_fusion.temperature + cooling_heat_amount / internal_fusion_heat_capacity, TCMB)
+	cooling_port.merge(cooling_removed)
```

Comment:
Bit uncomfortable with this. Using total_moles in this case isn't a huge deal, and when you remove it like this you open yourself up to some messy errors if the total changes after the proc. Would prefer this pattern of caching these values be kept to hot code and stuff that's simple

Reply:
Why would the total changing after the proc cause errors?

Reply:
It's not appropriate to cache the total moles of a gasmix when you want the total both before and after it "mutates", or changes.
Because of that, overusing this pattern can get a bit messy.
Also uh, `cooling_port_total_moles` doesn't need to exist, cause of remove_ratio